cpp/src/arrow/compute/exec/expression.cc

```diff
@@ -512,7 +511,31 @@ Result<Datum> ExecuteScalarExpression(const Expression& expr, const ExecBatch& i
       return MakeNullScalar(null());
     }
 
-    const Datum& field = input[param->index];
+    Datum field = input[param->indices[0]];
+    for (auto it = param->indices.begin() + 1; it != param->indices.end(); ++it) {
```

Comment:
Can this use `for (auto index : param-> indices)`?

Reply:
It skips the first index and I don't think it'd be clearer overall to avoid the explicit iterator.

Reply:
Ah, I missed the `+ 1`. Roger.